tests/test_json_schema.py

```diff
@@ -3459,3 +3492,124 @@ class OuterModel(BaseModel):
         'title': 'OuterModel',
         'type': 'object',
     }
+
+
+@pytest.mark.xfail(
+    reason=(
+        'We are calling __get_pydantic_json_schema__ too many times.'
+        ' The second time we analyze a model we get the CoreSchema from __pydantic_core_schema__.'
+        ' But then we proceed to append to the metadata json schema functions.'
+    )
+)
```

Comment:
This equivalent test shows that this is a bug in `main` that I just can't fix in this PR:

```python
from typing import Any, Dict, List

from pydantic import BaseModel
from pydantic.json_schema import JsonSchemaValue

calls: List[str] = []

class Model(BaseModel):
    @classmethod
    def __modify_schema__(cls, schema: Dict[str, Any]) -> JsonSchemaValue:
        calls.append('x')
        schema['title'] += 'x'
        return schema


class OuterModel(BaseModel):
    x: Model


assert OuterModel.model_json_schema()['$defs']['Model']['title'] == 'Modelxx'
assert calls == ['x', 'x']
```